Mathlib/Analysis/Convex/EGauge.lean

```diff
@@ -0,0 +1,202 @@
+/-
+Copyright (c) 2024 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Analysis.Seminorm
+
+/-!
+# The Minkowski functional, normed field version
+
+In tihs file we define `(egauge 𝕜 s ·)`
+to be the Minkowski functional (gauge) of the set `s`
+in a topological vector space `E` over a normed field `𝕜`,
+as a function `E → ℝ≥0∞`.
+
+It is defined as the infimum of the norms of `c : 𝕜` such that `x ∈ c • s`.
+In particular, for `𝕜 = ℝ≥0` this definition gives an `ℝ≥0∞`-valued version of `gauge`
+defined in `Mathlib/Analysis/Convex/Gauge.lean`.
+
+Currently, we can't reuse results about `egauge` for `gauge`,
+because we lack a theory of normed semifields.
+-/
+
+open Set Filter Metric
+open scoped Topology Pointwise ENNReal NNReal
+
+section SMul
+
+/-- The Minkowski functional for vector spaces over normed fields.
+Given a set `s` in a vector space over a normed field `𝕜`,
+`egauge s` is the functional which sends `x : E`
+to the infimum of `‖c‖₊` over `c` such that `x` belongs to `s` scaled by `c`.
+
+The definition only requires `𝕜` to have a `NNNorm` instance
+and `(· • ·) : 𝕜 → E → E` to be defined.
+This way the definition applies, e.g., to `𝕜 = ℝ≥0`.
+For `𝕜 = ℝ≥0`, the function is equal (up to conversion to `ℝ`)
+to the usual Minkowski functional defined in `gauge`. -/
```

Comment:
Is this result scheduled for a future PR? If it's simple, perhaps it belongs in this one.

Reply:
Which `import` should I add? Or move definitions to a new `Defs.lean` file so that the result can go to either file?

Reply:
I'm going to merge this, then open a new PR over the week-end moving defs to a new file and proving this theorem.